src/Abstractions/IUserRepository.cs

```diff
@@ -8,7 +7,14 @@ public interface IUserRepository
     {
 
         public List<User> FindAll();
-        public List<User> CreateOne(User user);
+        
+        public User CreateOne(User user);
+
+        public User? findOne(string userId);
+
+            public User? findOneByEmail(User userEmail);
```

Comment:
here the data type should be `string` because you are passing the email as a `string` not as a `User`